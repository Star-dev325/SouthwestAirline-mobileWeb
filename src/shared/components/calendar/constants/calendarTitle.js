export default {
  CALENDAR: 'Calendar',
  SELECT_DATE: 'Select Date',
  SELECT_DATES: 'Select Dates'
};
