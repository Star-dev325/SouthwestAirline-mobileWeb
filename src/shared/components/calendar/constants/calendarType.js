import { Dayjs } from 'dayjs';
// @flow
export const BOTH = 'BOTH';
export const DEPART = 'DEPART';
export const RETURN = 'RETURN';

export type DateRangeType = 'RETURN' | 'DEPART' | 'BOTH';

export type SelectedDatesType = {
  newOutboundDate: ?Dayjs,
  newInboundDate: ?Dayjs
};
