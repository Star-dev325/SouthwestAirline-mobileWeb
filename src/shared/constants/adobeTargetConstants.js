export default {
  HOME_PAGE_MBOX1_ID: 'swa_mweb_homeHero01',
  HOME_PAGE_MBOX2_ID: 'swa_mweb_homeHero02',
  RAPID_REWARDS_MBOX_ID: 'swa_mweb_rapidRewardsIndexpromoBanner03',
  PRICING_CHASE_MBOX_ID: 'swa_mweb_airBookingPriceFullChaseAd',
  PRICE_PROMO_MIDDLE1_MBOX_ID: 'mWebAirPricePromoMiddle01',
  PURCHASE_PAGE_MBOX_ID: 'mWebAirBookingPurchaseBtmAd1',
  GLOBAL_NAV_TOP_MBOX_ID: 'mWebGlobalNavTop',
  GLOBAL_NAV_PROMO1_MBOX_ID: 'mWebGlobalNavPromo1',
  OFFERS_PAGE_OFFER1_MBOX_ID: 'mWebOffersPageOffer1',
  OFFERS_PAGE_OFFER2_MBOX_ID: 'mWebOffersPageOffer2',
  EARLY_BIRD_PURCHASE_VISIBILITY_MBOX_ID: 'mWebAirPurchaseEarlyBirdBanner',
  EARLY_BIRD_PRICE_VISIBILITY_MBOX_ID: 'mWebAirPriceEarlyBirdBanner',
  RECENT_FLIGHT_DESTINATION_AIRPORT: 'recentFlightDestinationAirport',
  PURCHASE_PAYMENT_METHOD_MBOX_ID: 'mWebAirPurchasePaymentMethod',
  PURCHASE_PAYMENT_METHOD_UPLIFT_DISPLAY: 'mWebAirPurchasePaymentMethodUpliftDisplay',
  CONFIRMATION_PLACEMENT_MBOX_ID: 'mWebCheckoutConfirmationBottomPromo1',
  UPGRADED_BOARDING_PURCHASE_PROMO_TOP_01_MBOX_ID: 'mWebUpgradedBoardingPurchasePromoTop01',
  CHASE_AUDIENCE_WCM_CONTEXT: 'AudienceContextForWCM',
  CHASE_EXISTING_CARD_MEMBER: 'ecm',
  CHECK_IN_CONFIRMATION_UPGRADE_BUTTON: 'mWebCheckInConfirmationUpgradeButton',
  AIRPORTS_MULTI_GROUP_SELECT: 'airBookingIndexWidgetOrigDestMultiSelectGroupDisplay',
  PURCHASE_PAGE_SPLIT_PAY_DISPLAY: 'mWebAirPurchaseApplyPointsDisplay'
};
