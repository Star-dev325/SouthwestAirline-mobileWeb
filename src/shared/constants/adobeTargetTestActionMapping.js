import { setIsSplitPayVisible, setIsUpliftVisible } from 'src/airBooking/actions/airBookingActions';
import { saveDestinationAirport } from 'src/airports/actions/airportsActions';
import {
  setIsChaseExistingCardMember
} from 'src/chase/actions/chaseActions';
import AdobeTargetConstants from 'src/shared/constants/adobeTargetConstants';

const {
  RECENT_FLIGHT_DESTINATION_AIRPORT,
  PURCHASE_PAYMENT_METHOD_UPLIFT_DISPLAY,
  CHASE_AUDIENCE_WCM_CONTEXT,
  PURCHASE_PAGE_SPLIT_PAY_DISPLAY
} = AdobeTargetConstants;

export const adobeTargetTestActionMapping = () => ({
  [RECENT_FLIGHT_DESTINATION_AIRPORT]: saveDestinationAirport,
  [PURCHASE_PAYMENT_METHOD_UPLIFT_DISPLAY]: setIsUpliftVisible,
  [CHASE_AUDIENCE_WCM_CONTEXT]: setIsChaseExistingCardMember,
  [PURCHASE_PAGE_SPLIT_PAY_DISPLAY]: setIsSplitPayVisible
});
