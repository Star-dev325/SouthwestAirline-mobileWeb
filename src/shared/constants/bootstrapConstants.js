export default {
  APP_SETTINGS: 'app-settings/mobile-web-app-settings',
  APPLICATION_TOGGLES: 'application-toggles',
  CHASE_TTL_CONFIG_PATH: 'chase-ttl-config',
  CONTENT_PATH: 'bootstrap-content/en/content',
  ERROR_CODE_MAP_PATH: 'error-code-map',
  I18N_OVERRIDE_PATH: 'i18n-override',
  JWT_SETTINGS: 'jwt-settings',
  PAYMENT_OPTION_ORDER_PATH: 'payment-option-order',
  URLS_PATH: 'urls',
  WEBVIEW_ROUTES: 'webview-routes'
};
