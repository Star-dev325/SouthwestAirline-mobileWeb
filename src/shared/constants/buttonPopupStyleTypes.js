export const DEFAULT = 'default';
export const PRIMARY = 'primary';
export const DESTRUCTIVE = 'destructive';
