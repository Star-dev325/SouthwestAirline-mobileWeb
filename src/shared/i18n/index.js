import lapChildText from 'src/airBooking/i18n/lapChildText';
import airportListText from 'src/shared/i18n/airportListText';
import airportSelectorText from 'src/shared/i18n/airportSelectorText';
import billingInfoText from 'src/shared/i18n/billingInfoText';
import boardingInformationText from 'src/shared/i18n/boardingInformationText';
import boundSelectText from 'src/shared/i18n/boundSelectText';
import breakdownToggleNamesText from 'src/shared/i18n/breakdownToggleNamesText';
import buttonText from 'src/shared/i18n/buttonText';
import calendarText from 'src/shared/i18n/calendarText';
import carReservationText from 'src/shared/i18n/carReservationText';
import commonSharedText from 'src/shared/i18n/commonSharedText.js';
import contactMethodOptionsText from 'src/shared/i18n/contactMethodOptionsText';
import contactMethodText from 'src/shared/i18n/contactMethodText';
import earlyBirdInPathText from 'src/shared/i18n/earlyBirdInPathText';
import earlyBirdText from 'src/shared/i18n/earlyBirdText';
import errorPageText from 'src/shared/i18n/errorPageText';
import errorText from 'src/shared/i18n/errorText';
import flightProductText from 'src/shared/i18n/flightProductText';
import flightSegmentDetailsText from 'src/shared/i18n/flightSegmentDetailsText';
import flightsSummaryHeaderText from 'src/shared/i18n/flightsSummaryHeaderText';
import flightStatusText from 'src/shared/i18n/flightStatusText';
import footerText from 'src/shared/i18n/footerText';
import fundResultItemText from 'src/shared/i18n/fundResultItemText.js';
import hazmatDisclaimerText from 'src/shared/i18n/hazmatDisclaimerText';
import headerButtonText from 'src/shared/i18n/headerButtonText';
import labelStringText from 'src/shared/i18n/labelStringText';
import navGroupItemText from 'src/shared/i18n/navGroupItemText';
import passengerReservationTitleText from 'src/shared/i18n/passengerReservationTitleText';
import passportPageText from 'src/shared/i18n/passportPageText';
import paymentLinksText from 'src/shared/i18n/paymentLinksText';
import placeholderText from 'src/shared/i18n/placeholderText';
import priceLineTitlesText from 'src/shared/i18n/priceLineTitlesText';
import promoCodesText from 'src/shared/i18n/promoCodesText';
import purchaseSummaryFormText from 'src/shared/i18n/purchaseSummaryFormText';
import recentSearchesText from 'src/shared/i18n/recentSearchesText';
import redressAndKnownTravelerFieldsText from 'src/shared/i18n/redressAndKnownTravelerFieldsText';
import refundMethodsText from 'src/shared/i18n/refundMethodsText';
import repricingText from 'src/shared/i18n/repricingText';
import reservationPassengerText from 'src/shared/i18n/reservationPassengerText';
import reviewFooterText from 'src/shared/i18n/reviewFooterText';
import savedCreditCardsTitleText from 'src/shared/i18n/savedCreditCardsTitleText';
import selectPassengersPageText from 'src/shared/i18n/selectPassengersPageText';
import shareItineraryText from 'src/shared/i18n/shareItineraryText';
import specialAssistanceText from 'src/shared/i18n/specialAssistanceText';
import spinnerText from 'src/shared/i18n/spinnerText';
import summaryInsufficientText from 'src/shared/i18n/summaryInsufficientText';
import tripBookedText from 'src/shared/i18n/tripBookedText';

const shared = {
  ...airportListText,
  ...airportSelectorText,
  ...billingInfoText,
  ...boardingInformationText,
  ...boundSelectText,
  ...breakdownToggleNamesText,
  ...buttonText,
  ...calendarText,
  ...carReservationText,
  ...commonSharedText,
  ...contactMethodOptionsText,
  ...contactMethodText,
  ...earlyBirdInPathText,
  ...earlyBirdText,
  ...errorPageText,
  ...errorText,
  ...flightProductText,
  ...flightSegmentDetailsText,
  ...flightsSummaryHeaderText,
  ...flightStatusText,
  ...footerText,
  ...fundResultItemText,
  ...hazmatDisclaimerText,
  ...headerButtonText,
  ...labelStringText,
  ...lapChildText,
  ...navGroupItemText,
  ...passengerReservationTitleText,
  ...passportPageText,
  ...paymentLinksText,
  ...placeholderText,
  ...priceLineTitlesText,
  ...promoCodesText,
  ...purchaseSummaryFormText,
  ...recentSearchesText,
  ...redressAndKnownTravelerFieldsText,
  ...refundMethodsText,
  ...repricingText,
  ...reservationPassengerText,
  ...reviewFooterText,
  ...savedCreditCardsTitleText,
  ...selectPassengersPageText,
  ...shareItineraryText,
  ...specialAssistanceText,
  ...spinnerText,
  ...summaryInsufficientText,
  ...tripBookedText
};

export default shared;
