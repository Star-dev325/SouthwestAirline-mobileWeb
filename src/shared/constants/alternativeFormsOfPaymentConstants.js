import { FIELD_ERROR_MESSAGE } from 'src/shared/form/constants/validationErrorTypes';
import i18n from '@swa-ui/locale';

export const EXTERNAL_PAYMENT_PAGE_URL = '/payment/external';

export const PAYMENT_METHODS = {
  APPLE_PAY: 'ApplePay',
  UPLIFT: 'PayMonthly'
};

export const INITIAL_AVAILABILITY = {
  paymentMethod: '',
  isAvailable: false,
  isActive: false,
  hasError: false,
  lastUpdateFailed: false,
  parameters: {},
  shouldDisplay: false
};

export const INITIAL_TRIP_INFO = {
  air_reservations: [{}],
  travelers: [{}]
};

export const validationTransformer = {
  firstName: {
    ceptorParam: 'givenName',
    requiredErrorMessage: i18n('SHARED__ERROR_MESSAGES__PASSENGER_FIRST_NAME_VALID')
  },
  lastName: {
    ceptorParam: 'familyName',
    requiredErrorMessage: i18n('SHARED__ERROR_MESSAGES__PASSENGER_LAST_NAME_VALID')
  },
  addressLine1: {
    ceptorParam: 'addressLines',
    requiredErrorMessage: i18n('SHARED__ERROR_MESSAGES__ADDRESS_LINE_1_LENGTH')
  },
  addressLine2: {
    ceptorParam: 'addressLines'
  },
  city: {
    ceptorParam: 'locality',
    requiredErrorMessage: i18n('SHARED__ERROR_MESSAGES__CITY')
  },
  stateProvinceRegion: {
    ceptorParam: 'administrativeArea',
    requiredErrorMessage: i18n('SHARED__ERROR_MESSAGES__STATE_PROVINCE_REGION')
  },
  zipOrPostalCode: {
    ceptorParam: 'postalCode',
    requiredErrorMessage: i18n('SHARED__ERROR_MESSAGES__ZIP_POSTAL_CODE')
  },
  isoCountryCode: {
    ceptorParam: 'countryCode'
  }
};

export const invalidUsStateError = {
  stateProvinceRegion: {
    msg: i18n('SHARED__ERROR_MESSAGES__STATE_PROVINCE_REGION'),
    type: FIELD_ERROR_MESSAGE
  }
};

export const APPLICATION_TYPES = {
  AIR_BOOKING: 'air/booking',
  AIR_CHANGE: 'air/change',
  AIR_UPGRADE: 'air/upgrade',
  COMPANION: 'companion',
  EARLYBIRD: 'earlybird',
  SAME_DAY: 'same-day',
  UPGRADED_BOARDING: 'upgraded-boarding'
};

export const APPLICATION_TYPES_UPLIFT = {
  'air/booking': 'air-booking',
  'air/change': 'air-change',
  'air/upgrade': 'air/upgrade',
  companion: 'companion',
  earlybird: 'earlybird',
  'same-day': 'same-day',
  'upgraded-boarding': 'upgraded-boarding'
};

export const PRICE_TYPES = {
  UP_TRIP_TOTAL: 'up-trip-total',
  UP_EARLY_BIRD_CHECK_IN: 'up-early-bird-check-in'
};

export const DEFAULT_ERROR_AFP_CODES_TO_DISPLAY = ['811', '813'];
export const DEFAULT_UPLIFT_TIME_TO_TRAVEL_HOURS_LIMIT = 24;
export const DEFAULT_UPLIFT_PAX_AGE_LIMIT = 18;
