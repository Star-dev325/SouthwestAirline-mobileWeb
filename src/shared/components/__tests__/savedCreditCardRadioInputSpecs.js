import React from 'react';
import { mount, shallow } from 'enzyme';
import sinonModule from 'sinon';
import SavedCreditCardRadioInput from 'src/shared/components/savedCreditCardRadioInput';
import PaymentSavedCreditCardBuilder from 'test/builders/model/paymentSavedCreditCardBuilder';
import {
  RAPID_REWARDS_VISA_ID,
  NEW_CREDIT_CARD_ID,
  PAY_PAL_CARD_ID,
  APPLE_PAY_CARD_ID,
  UPLIFT_CARD_ID
} from 'src/shared/constants/creditCardConstants';
import CeptorWrapper from 'src/shared/helpers/ceptorWrapper';
import { PRICE_TYPES } from 'src/shared/constants/alternativeFormsOfPaymentConstants';
import i18n from '@swa-ui/locale';

const sinon = sinonModule.sandbox.create();

describe('SavedCreditCardRadioInput', () => {
  let wrapper;
  let onClickStub;
  let creditCardInfo;
  let getExtensionStub;

  beforeEach(() => {
    getExtensionStub = sinon.stub(CeptorWrapper, 'getExtension');
  });

  afterEach(() => {
    sinon.restore();
  });

  context('render', () => {
    beforeEach(() => {
      creditCardInfo = new PaymentSavedCreditCardBuilder().build();

      wrapper = createComponent({}, false);
    });

    it('should pass correct props to CreditCardRadioInput', () => {
      const creditCardRadioInput = wrapper.find('CreditCardRadioInput');

      expect(creditCardRadioInput).to.have.prop('onClick', onClickStub);
      expect(creditCardRadioInput).to.have.prop('selected', false);
      expect(creditCardRadioInput).to.have.prop('savedCreditCardId', '1-ENKS4K');
      expect(creditCardRadioInput).to.have.prop('name', 'VISA 9999');
      expect(creditCardRadioInput).to.have.prop('type', 'VISA');
      expect(creditCardRadioInput).to.have.prop('showRadioButton', true);
    });

    it('should call onClick prop with new credit card id when click on input', () => {
      wrapper.find('CreditCardRadioInput').simulate('click');

      expect(onClickStub).to.be.calledWith('1-ENKS4K');
    });
  });

  it('should call ceptor wrapper getExtension and addInfo function when Uplift', () => {
    creditCardInfo = {
      savedCreditCardId: UPLIFT_CARD_ID,
      type: 'UPLIFT',
      name: 'Pay Monthly'
    };
    const addInfoStub = sinon.stub();

    getExtensionStub.returns({ addInfo: addInfoStub });
    wrapper = createComponent(
      { additionalInfoMessage: 'Pay Monthly from $XX/mo', additionalInfoLink: 'Learn More' },
      false
    );

    expect(getExtensionStub).to.have.been.called;
    expect(addInfoStub).to.have.been.calledWith(sinon.match.defined, PRICE_TYPES.UP_TRIP_TOTAL);
  });

  it('should show Just added! text when justAdded is true', () => {
    creditCardInfo = {
      savedCreditCardId: RAPID_REWARDS_VISA_ID,
      type: 'INSTANT_CREDIT_RAPID_REWARDS_VISA',
      name: i18n('SHARED__SAVED_CREDIT_CARDS_TITLE__RAPID_REWARDS_VISA')
    };
    wrapper = createComponent({
      justAdded: true
    });

    expect(wrapper).to.matchSnapshot();
  });

  it('should show additional info container when type is ghost card and its expired', () => {
    creditCardInfo = {
      savedCreditCardId: 'First Ghost Card',
      type: 'GHOST_CARD',
      name: 'First Ghost Card',
      isExpired: true
    };
    wrapper = createComponent();

    expect(wrapper).to.matchSnapshot();
  });

  it('should show additional info container for ghost card if additionalInfoMessage has value', () => {
    wrapper = createComponent({
      additionalInfoMessage: i18n('AIR_BOOKING__CORPORATE_BOOKING__TAP_TO_SELECT_A_DIFFERENT_CARD')
    });

    expect(wrapper).to.matchSnapshot();
  });

  it('should show last 4 digits when it is VISA credit card', () => {
    creditCardInfo = {
      savedCreditCardId: '1-YNA593',
      type: 'VISA',
      name: 'VISA Card'
    };
    wrapper = createComponent();

    expect(wrapper).to.matchSnapshot();
  });

  it('should not show last 4 digits when it is a chase credit card', () => {
    creditCardInfo = {
      savedCreditCardId: RAPID_REWARDS_VISA_ID,
      type: 'INSTANT_CREDIT_RAPID_REWARDS_VISA',
      name: i18n('SHARED__SAVED_CREDIT_CARDS_TITLE__RAPID_REWARDS_VISA')
    };
    wrapper = createComponent();

    expect(wrapper).to.matchSnapshot();
  });

  it('should not show last 4 digits when it is a new credit card', () => {
    creditCardInfo = {
      savedCreditCardId: NEW_CREDIT_CARD_ID,
      type: 'NEW',
      name: i18n('SHARED__SAVED_CREDIT_CARDS_TITLE__NEW_CREDIT_CARD')
    };
    wrapper = createComponent();

    expect(wrapper).to.matchSnapshot();
  });

  it('should not show last 4 digits when new credit card has been added', () => {
    creditCardInfo = {
      savedCreditCardId: 'MASTERCARD',
      type: 'NEW',
      name: 'MasterCard'
    };
    wrapper = createComponent();

    expect(wrapper).to.matchSnapshot();
  });

  it('should not show last 4 digits when its new disabled credit card', () => {
    creditCardInfo = {
      savedCreditCardId: 'MASTERCARD',
      type: 'NEW_DISABLED',
      name: 'MasterCard'
    };
    wrapper = createComponent();

    expect(wrapper).to.matchSnapshot();
  });

  it('should not show last 4 digits when it is a paypal card', () => {
    creditCardInfo = {
      savedCreditCardId: PAY_PAL_CARD_ID,
      type: 'PAYPAL',
      name: 'Use PayPal'
    };
    wrapper = createComponent();

    expect(wrapper).to.matchSnapshot();
  });

  it('should not show last 4 digits when it is an apple pay card', () => {
    creditCardInfo = {
      savedCreditCardId: APPLE_PAY_CARD_ID,
      type: 'APPLE_PAY',
      name: 'Use Apple Pay'
    };
    wrapper = createComponent();

    expect(wrapper).to.matchSnapshot();
  });

  it('should not show last 4 digits when it is ghost card', () => {
    creditCardInfo = {
      savedCreditCardId: 'First Ghost Card',
      type: 'GHOST_CARD',
      name: 'First Ghost Card'
    };
    wrapper = createComponent();

    expect(wrapper).to.matchSnapshot();
  });

  it('should not show additional info container for ghost card if additionalInfoMessage has no value', () => {
    wrapper = createComponent();

    expect(wrapper).to.not.contain.text(i18n('AIR_BOOKING__CORPORATE_BOOKING__TAP_TO_SELECT_A_DIFFERENT_CARD'));
  });

  it('should show theÂ Paypal text when it is a paypal card', () => {
    creditCardInfo = {
      savedCreditCardId: PAY_PAL_CARD_ID,
      type: 'PAYPAL',
      name: 'Use PayPal'
    };
    wrapper = createComponent();

    expect(wrapper).to.matchSnapshot();
  });

  it('should show additionalInfoLink text when passed in', () => {
    creditCardInfo = {
      savedCreditCardId: UPLIFT_CARD_ID,
      type: 'UPLIFT',
      name: 'Pay Monthly'
    };
    wrapper = createComponent({ additionalInfoLink: 'Learn More' });

    expect(wrapper).to.matchSnapshot();
  });

  const createComponent = (props = {}, shouldShallow = true) => {
    onClickStub = sinon.stub();

    const mergedProps = {
      creditCardInfo,
      selected: false,
      onClick: onClickStub,
      showRadioButton: true,
      ...props
    };

    return shouldShallow
      ? shallow(<SavedCreditCardRadioInput {...mergedProps} />)
      : mount(<SavedCreditCardRadioInput {...mergedProps} />);
  };
});
