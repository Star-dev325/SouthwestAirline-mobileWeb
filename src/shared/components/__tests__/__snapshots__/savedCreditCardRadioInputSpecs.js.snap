// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedCreditCardRadioInput should not show last 4 digits when it is a chase credit card 1`] = `
<CreditCardRadioInput
  name="RAPID REWARDS® VISA"
  onClick={[Function]}
  savedCreditCardId="RAPID_REWARDS_VISA_ID"
  selected={false}
  type="INSTANT_CREDIT_RAPID_REWARDS_VISA"
/>
`;

exports[`SavedCreditCardRadioInput should not show last 4 digits when it is a new credit card 1`] = `
<CreditCardRadioInput
  name="Use new credit card"
  onClick={[Function]}
  savedCreditCardId="NEW_CREDIT_CARD_ID"
  selected={false}
  type="NEW"
/>
`;

exports[`SavedCreditCardRadioInput should not show last 4 digits when it is a paypal card 1`] = `
<CreditCardRadioInput
  name="Use PayPal"
  onClick={[Function]}
  savedCreditCardId="PAY_PAL_CARD_ID"
  selected={false}
  type="PAYPAL"
>
  <div
    className="flex gray5 additional-info-container"
  >
    <div
      className="mr1"
    >
      Buy now or pay later
    </div>
  </div>
</CreditCardRadioInput>
`;

exports[`SavedCreditCardRadioInput should not show last 4 digits when it is an apple pay card 1`] = `
<CreditCardRadioInput
  name="Use Apple Pay"
  onClick={[Function]}
  savedCreditCardId="APPLE_PAY_CARD_ID"
  selected={false}
  type="APPLE_PAY"
/>
`;

exports[`SavedCreditCardRadioInput should not show last 4 digits when it is ghost card 1`] = `
<CreditCardRadioInput
  name="First Ghost Card"
  onClick={[Function]}
  savedCreditCardId="First Ghost Card"
  selected={false}
  type="GHOST_CARD"
/>
`;

exports[`SavedCreditCardRadioInput should not show last 4 digits when its new disabled credit card 1`] = `
<CreditCardRadioInput
  name="MasterCard"
  onClick={[Function]}
  savedCreditCardId="MASTERCARD"
  selected={false}
  type="NEW_DISABLED"
/>
`;

exports[`SavedCreditCardRadioInput should not show last 4 digits when new credit card has been added 1`] = `
<CreditCardRadioInput
  name="MasterCard"
  onClick={[Function]}
  savedCreditCardId="MASTERCARD"
  selected={false}
  type="NEW"
/>
`;

exports[`SavedCreditCardRadioInput should show Just added! text when justAdded is true 1`] = `
<CreditCardRadioInput
  name="RAPID REWARDS® VISA"
  onClick={[Function]}
  savedCreditCardId="RAPID_REWARDS_VISA_ID"
  selected={false}
  type="INSTANT_CREDIT_RAPID_REWARDS_VISA"
>
  <div>
    <div
      className="small green bold mt2"
    >
      Just added!
    </div>
  </div>
</CreditCardRadioInput>
`;

exports[`SavedCreditCardRadioInput should show additional info container for ghost card if additionalInfoMessage has value 1`] = `
<CreditCardRadioInput
  name="First Ghost Card"
  onClick={[Function]}
  savedCreditCardId="First Ghost Card"
  selected={false}
  type="GHOST_CARD"
>
  <div
    className="flex additional-info-container gray5"
  >
    <div
      className="mr1"
    >
      Tap to select a different corporate card
    </div>
  </div>
  <div
    className="flex gray5 additional-info-container"
  >
    <div
      className="red bold"
      data-qa="expired-credit-card"
    >
      EXPIRED
    </div>
  </div>
</CreditCardRadioInput>
`;

exports[`SavedCreditCardRadioInput should show additional info container when type is ghost card and its expired 1`] = `
<CreditCardRadioInput
  name="First Ghost Card"
  onClick={[Function]}
  savedCreditCardId="First Ghost Card"
  selected={false}
  type="GHOST_CARD"
>
  <div
    className="flex gray5 additional-info-container"
  >
    <div
      className="red bold"
      data-qa="expired-credit-card"
    >
      EXPIRED
    </div>
  </div>
</CreditCardRadioInput>
`;

exports[`SavedCreditCardRadioInput should show additionalInfoLink text when passed in 1`] = `
<CreditCardRadioInput
  name="Pay Monthly"
  onClick={[Function]}
  savedCreditCardId="UPLIFT_CARD_ID"
  selected={false}
  type="UPLIFT"
>
  <div
    className="flex medium pblue"
  >
    <span>
      Learn More
    </span>
  </div>
</CreditCardRadioInput>
`;

exports[`SavedCreditCardRadioInput should show last 4 digits when it is VISA credit card 1`] = `
<CreditCardRadioInput
  name="VISA Card"
  onClick={[Function]}
  savedCreditCardId="1-YNA593"
  selected={false}
  type="VISA"
>
  <div
    className="flex gray5 additional-info-container"
  >
    <div
      className="mr1"
    >
      Last 4 digits: 
    </div>
    <div
      className="bold"
    />
  </div>
</CreditCardRadioInput>
`;

exports[`SavedCreditCardRadioInput should show the Paypal text when it is a paypal card 1`] = `
<CreditCardRadioInput
  name="Use PayPal"
  onClick={[Function]}
  savedCreditCardId="PAY_PAL_CARD_ID"
  selected={false}
  type="PAYPAL"
>
  <div
    className="flex gray5 additional-info-container"
  >
    <div
      className="mr1"
    >
      Buy now or pay later
    </div>
  </div>
</CreditCardRadioInput>
`;
